### ISSUE TITLE
Not possible to override package manager

### ISSUE THREAD

**Issue by jmtd** [cekit/cekit#462]
with cekit develop (4ff6bd5ad9bfc2c058a8a68cfc83fa283ddc3640),
if I make the following changes to an image that builds correctly

```
@@ -1,6 +1,6 @@
 schema_version: 1
 
-from: "ubi8:latest"
+from: "ubi8-minimal:latest"
 name: &name "openjdk/openjdk-11-rhel8"
 version: &version "1.0"
 description: "Source To Image (S2I) image for Red Hat OpenShift providing OpenJDK 11"
@@ -33,6 +33,7 @@ osbs:
 #openjdk-11-rhel8
 
 packages:
+  manager: microdnf
   content_sets:
     x86_64:
       - rhel-8-for-x86_64-baseos-rpms
```

cekit build fails, with the errors suggesting that yum was still being tried for the "makecache" stage:

```
2019-04-01 14:59:24,139 cekit        INFO     Docker: Step 3/84 : COPY repos/content_sets_odcs.repo /etc/yum.repos.d/
2019-04-01 14:59:24,339 cekit        INFO     Docker: ---> e51ad23b31e2
2019-04-01 14:59:24,347 cekit        INFO     Docker: Removing intermediate container f20c66356d55
2019-04-01 14:59:24,348 cekit        INFO     Docker: Step 4/84 : RUN yum makecache
2019-04-01 14:59:24,376 cekit        INFO     Docker: ---> Running in 573d2f337242
2019-04-01 14:59:24,381 cekit        INFO     Docker: {}
2019-04-01 14:59:24,381 cekit        INFO     Docker: 
2019-04-01 14:59:24,657 cekit        INFO     Docker: /bin/sh: yum: command not found
2019-04-01 14:59:24,840 cekit        ERROR    You can look inside the failed image by running 'docker run --rm -ti 499dee25ce43 bash'
2019-04-01 14:59:24,842 cekit        ERROR    ('Image build failed, see logs above.', CekitError("Image build failed: 'The command '/bin/sh -c yum makecache' returned a non-zero code: 127'",))
Traceback (most recent call last):
  File "/home/ce/cekit/develop/lib/python2.7/site-packages/cekit/cli.py", line 336, in run
    command.execute()
  File "/home/ce/cekit/develop/lib/python2.7/site-packages/cekit/builder.py", line 60, in execute
    self.run()
  File "/home/ce/cekit/develop/lib/python2.7/site-packages/cekit/builders/docker_builder.py", line 183, in run
    image_id = self._build_with_docker(docker_client)
  File "/home/ce/cekit/develop/lib/python2.7/site-packages/cekit/builders/docker_builder.py", line 149, in _build_with_docker
    raise CekitError(msg, ex)
CekitError: ('Image build failed, see logs above.', CekitError("Image build failed: 'The command '/bin/sh -c yum makecache' returned a non-zero code: 127'",))
```

**Comment by goldmann** [comment]
Thanks for this report, I'll take a look at this!

**Comment by goldmann** [comment]
I've added a check to the tests, run them but I couldn't reproduce it, looking again.

**Comment by goldmann** [comment]
Do you use some overrides by chance? The only thing that comes to my mind is that this default is applied in some higher (or lower?) level descriptor: https://github.com/cekit/cekit/blob/develop/cekit/descriptor/packages.py#L23

**Comment by goldmann** [comment]
@jmtd Take a look at this PR which adds a few more checks: https://github.com/cekit/cekit/pull/464. There I specifically added a test for the execution of `makecache` with different package managers. Tests pass.

I'm not sure why it's failing in your environment.

**Comment by jmtd** [comment]
> Do you use some overrides by chance?

I should have specified more clearly, that diff applies to `rhel8-jdk11-overrides.yaml`, I am executing `cekit build --overrides-file rhel8-jdk11-overrides.yaml docker`. But I agree that I think there must be an environmental issue here, I'm continuing to dig.

After updating to 6077bb926a856ae5bb634c9742caa179d8f60cf0 I still get the same result. could the content_sets overriding be ignoring the manager setting?

**Comment by jmtd** [comment]
Some minor progress. If I copy the `manager: microdnf` into the `image.yaml`, instead of the override, then it takes effect (and I get a different error, #465). So this appears to be a problem with overrides.

The image sources I am building are git@github.com:jmtd/openjdk.git branch rhel8-2k19 with git@github.com:jmtd/cct_module.git cloned within, branch rhel82k19, and the diff from the first post in this issue applied on top to the parent openjdk repository. I've rebuilt cekit in a fresh virtualenv and prepared fresh clones of the image repositories to try and rule out various sources of environmental pollution. There are no cekit-related files outside of the virtualenvs I am using (nothing remains in /usr etc from old package installs)

**Comment by jmtd** [comment]
yes the overrides are a problem. this change to the test demonstrates it

```
diff --git a/tests/test_dockerfile.py b/tests/test_dockerfile.py
index 27366f4..7453b90 100644
--- a/tests/test_dockerfile.py
+++ b/tests/test_dockerfile.py
@@ -302,8 +302,8 @@ def test_default_package_manager(tmpdir):
 def test_dockerfile_custom_package_manager_with_overrides(tmpdir):
     target = str(tmpdir.mkdir('target'))
 
-    generate(target, ['-v', 'build', '--overrides', '{"packages": {"install": ["b"]}}', '--dry-run', 'docker'],
-             descriptor={'packages': {'manager': 'microdnf',
+    generate(target, ['-v', 'build', '--overrides', '{"packages": {"manager": "microdnf", "install": ["b"]}}', '--dry-run', 'docker'],
+             descriptor={'packages': {'manager': 'yum',
                                       'repositories': [{'name': 'foo',
                                                         'rpm': 'foo-repo.rpm'}],
                                       'install': ['a']},

```

**Comment by goldmann** [comment]
Thank you! I'm starting work on this.

**Comment by goldmann** [comment]
A proposed fix is available in #467. Thank you very much for digging deep and showing me what is the issue exactly!
